src/tests/libecalc/dto/results/model/test_requested_pressures_compressors.py

```diff
@@ -116,3 +121,36 @@ def test_requested_pressures_compressor_system_temporal_model(result: EcalcModel
     assert requested_inlet_pressure_train2_upgr == 45
     assert requested_outlet_pressure_train1_upgr == 240
     assert requested_outlet_pressure_train2_upgr == 245
+
+
+def test_model_timestep_not_in_energy_usage_model(compressor_systems_and_compressor_train_temporal_dto):
+    graph = compressor_systems_and_compressor_train_temporal_dto.ecalc_model.get_graph()
+    compressor_system = graph.get_node("compressor_system_variable_speed_compressor_trains_multiple_pressures")
+
+    energy_usage_model = compressor_system.energy_usage_model
+
+    variables = compressor_systems_and_compressor_train_temporal_dto.variables
+    energy_calculator = EnergyCalculator(graph=graph)
+    consumer_results = energy_calculator.evaluate_energy_usage(variables)
+    operational_settings_used = consumer_results[
+        "compressor_system_variable_speed_compressor_trains_multiple_pressures"
+    ].component_result.operational_settings_used
+
+    # Set some model time steps different from dates in temporal model
+    model_timesteps = [
+        datetime(2018, 1, 1),
+        datetime(2018, 6, 23, 17, 20, 50),
+        datetime(2018, 7, 5, 12, 20, 50),
+        datetime(2022, 1, 1),
+    ]
+
+    operational_settings_used.timesteps = model_timesteps
+
+    # Ensure that method do not fail if model timesteps do not correspond exactly to dates in temporal model
```

Comment:
also check that the evaluated/fetched pressure values are those one expect to get at the different timesteps?

Reply:
Updated to check the pressures as well.